plugin/src/main/kotlin/tanvd/grazi/language/Lang.kt

```diff
@@ -1,47 +1,58 @@
-    CANADIAN_ENGLISH(CanadianEnglish(), GraziBundle.langConfig("en.enabled"), GraziBundle.langConfig("en.disabled")),
-    RUSSIAN(Russian(), GraziBundle.langConfig("ru.enabled")),
-    PERSIAN(Persian()),
-    FRENCH(French()),
-    GERMANY_GERMAN(GermanyGerman()),
-    AUSTRIAN_GERMAN(AustrianGerman()),
-    POLISH(Polish()),
-    ITALIAN(Italian()),
-    DUTCH(Dutch()),
-    PORTUGAL_PORTUGUESE(PortugalPortuguese()),
-    BRAZILIAN_PORTUGUESE(BrazilianPortuguese()),
-    CHINESE(Chinese()),
-    GREEK(Greek()),
-    JAPANESE(Japanese()),
-    ROMANIAN(Romanian()),
-    SLOVAK(Slovak()),
-    SPANISH(Spanish()),
-    UKRAINIAN(Ukrainian());
+                private val disabledRules: Set<String> = emptySet(),
+                private val disabledCategories: Set<String> = emptySet()) {
+    BRITISH_ENGLISH("English (GB)", "en", "BritishEnglish",
+            GraziBundle.langConfig("en.rules.enabled"), GraziBundle.langConfig("en.rules.disabled"), GraziBundle.langConfig("en.categories.disabled")),
+    AMERICAN_ENGLISH("English (US)", "en", "AmericanEnglish",
+            GraziBundle.langConfig("en.rules.enabled"), GraziBundle.langConfig("en.rules.disabled"), GraziBundle.langConfig("en.categories.disabled")),
+    CANADIAN_ENGLISH("English (Canadian)", "en", "CanadianEnglish",
+            GraziBundle.langConfig("en.rules.enabled"), GraziBundle.langConfig("en.rules.disabled"), GraziBundle.langConfig("en.categories.disabled")),
+    RUSSIAN("Russian", "ru", "Russian", GraziBundle.langConfig("ru.rules.enabled")),
+    PERSIAN("Persian", "fa", "Persian"),
+    FRENCH("French", "fr", "French"),
+    GERMANY_GERMAN("German (Germany)", "de", "GermanyGerman"),
+    AUSTRIAN_GERMAN("German (Austria)", "de", "AustrianGerman"),
+    POLISH("Polish", "pl", "Polish"),
+    ITALIAN("Italian", "it", "Italian"),
+    DUTCH("Dutch", "nl", "Dutch"),
+    PORTUGAL_PORTUGUESE("Portuguese (Portugal)", "pt", "PortugalPortuguese"),
+    BRAZILIAN_PORTUGUESE("Portuguese (Brazil)", "pt", "BrazilianPortuguese"),
+    CHINESE("Chinese", "zh", "Chinese"),
+    GREEK("Greek", "el", "Greek"),
+    JAPANESE("Japanese", "ja", "Japanese"),
+    ROMANIAN("Romanian", "ro", "Romanian"),
+    SLOVAK("Slovak", "sk", "Slovak"),
+    SPANISH("Spanish", "es", "Spanish"),
+    UKRAINIAN("Ukrainian", "uk", "Ukrainian");
 
     companion object {
-        operator fun get(lang: Language): Lang? = values().find { it.shortCode == lang.shortCode }
+        operator fun get(lang: Language): Lang? = values().find { lang.name == it.displayName }
         operator fun get(code: String): Lang? = values().find { it.shortCode == code }
 
-        val sortedValues = values().sortedBy { it.displayName }
+        fun sortedValues() = values().sortedBy(Lang::displayName)
     }
 
-    val shortCode = jLanguage.shortCode!!
-
-    val displayName = jLanguage.name!!
+    val jLanguage: Language?
+        get() = tryRun {
+            if (ApplicationManager.getApplication().isUnitTestMode) {
```

Comment:
Probably we can move logic of correct loading of class to `GraziPlugin`? Something like `GraziPlugin.loadClass`

Reply:
Done